dev/source/docs/editing-prs.rst

```diff
@@ -0,0 +1,53 @@
+.. _editing-prs:
+
+======================================
+How to Edit Someone Else's PR Directly
+======================================
+
+Occasionally, it is desirable to make some small changes to someone else's submitted PR in order to facilitate its readiness for merging. This describes one procedure for doing this.
+
+Assumptions
+===========
+
+It is assumed that you have worked on the GitHub repository for ArduPilot before and have cloned a fork of the repository locally.
+
+Procedure
+=========
+
+First, determine if you have been given permission to modify the PR. If so, then at the bottom of the PR on GitHub.com you will see :
+
+"Add more commits by pushing to the ``xyz`` branch on ``username/ardupilot``"
+
+First set up a remote to username's repository, in this case we will call it ``tempremote`` : 
+
+git remote add ``tempremote`` https://github.com/``username/ardupilot``.git
+
+this sets up ``tempremote`` to point to the originators repository on GitHub.
+
+To pull the PR branch ``xyz`` down to your local computer's repository in order to modify it:
+
+git fetch ``tempremote``  ``xyz``
+
+which will copy the PR's branch into a local buffer.
+
+Then create a local branch named the same as the remote branch, ``xyz`` and which is a copy of the PR
+
+git checkout -b ``xyz`` FETCH_HEAD 
+
+
+Next make your mods and commit them with a commit message explaining your changes
+
+Then push your commit to the originators PR branch:
+
+git push ``tempremote``
+
+You should add a comment to the PR request comment box informing the originator that you have made a change to the PR in his branch.
+
+Before the PR is merged into the Ardupilot master, the PR originator should "squash" the new commit into his base commit to provide a single commit and commit message for the merge into Ardupilot.
+
+
+You could also "squash" your commit into the originator's before pushing to his repository yourself and "force push" it to overwrite his repository to appear as if a single commit has been made, if the originator is not familiar with "squashing" or no longer available by pushing with the -f option.
```

Comment:
By the way, we have this existing page which talks a bit about squashing: https://ardupilot.org/dev/docs/git-interactive-rebase.html.  Maybe a link would be helpful.. your choice though

Reply:
will add info on that....more appropriate since even intermediate level guys may not know much about squashing....I need to add info on  commit splitting to the main pages on contributing and probably should add that also here...